lib/flapjack/data/entity_check.rb

```diff
@@ -649,6 +651,8 @@ def update_state(new_state, options = {})
 
           # Even if this isn't a state change, we need to update the current state
           # hash summary and details (as they may have changed)
+          multi.hset("check:#{@key}", 'initial_failure_delay', (initial_failure_delay || Flapjack::DEFAULT_INITIAL_FAILURE_DELAY))
+          multi.hset("check:#{@key}", 'repeat_failure_delay', (repeat_failure_delay || Flapjack::DEFAULT_REPEAT_FAILURE_DELAY))
           multi.hset("check:#{@key}", 'summary', (summary || ''))
```

Comment:
This will reset the delay values unless every event contains a reinforcement of the non-standard delay period; would it make more sense to remain at the last set value if the event doesn't specify anything?